client/definitions.go

```diff
@@ -317,19 +260,26 @@ func GetDefinitionFromStore(
 			return host, true, nil
 		}
 
-		hostIDBase := strings.Split(host.Schema.Root, "-")[0]
-		targetID := fmt.Sprintf("%s-%s", hostIDBase, typedKind.RelativeID)
-
 		cols, err := store.GetCollections(ctx, CollectionFetchOptions{
-			CollectionID: immutable.Some(targetID),
+			CollectionSetID: immutable.Some(host.Version.CollectionSet.Value().CollectionSetID),
 		})
-		if len(cols) == 0 || err != nil {
+		if err != nil {
 			return CollectionDefinition{}, false, err
 		}
-		def := cols[0].Definition()
-		def.Version = CollectionVersion{}
 
-		return def, true, nil
+		for _, col := range cols {
+			if col.Version().CollectionID == host.Version.CollectionID {
+				continue
+			}
+
+			if col.Version().CollectionSet.Value().CollectionSetID != host.Version.CollectionSet.Value().CollectionSetID {
+				continue
+			}
+
+			if fmt.Sprint(col.Version().CollectionSet.Value().RelativeID) == typedKind.RelativeID {
```

Comment:
`typedKind.RelativeID` should really be an `Option[int]`, but it is an easy change to skip in this PR.  The horrible string cast can be removed then.  Let me know if you want a ticket opened for this.

Reply:
Yes please create a ticket and add a TODO comment with link to the ticket. This could be a good first issue for new devs or community devs to take on if we don't do it now/shortly.